src/main/java/io/github/opencubicchunks/cubicchunks/chunk/biome/ProtoColumnBiomeContainer.java

```diff
@@ -0,0 +1,101 @@
+package io.github.opencubicchunks.cubicchunks.chunk.biome;
+
+import io.github.opencubicchunks.cubicchunks.chunk.IBigCube;
+import io.github.opencubicchunks.cubicchunks.mixin.access.common.BiomeContainerAccess;
+import io.github.opencubicchunks.cubicchunks.server.CubicLevelHeightAccessor;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import net.minecraft.core.IdMap;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.LevelHeightAccessor;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.BiomeSource;
+import net.minecraft.world.level.chunk.ChunkBiomeContainer;
+import org.jetbrains.annotations.Nullable;
+
+public class ProtoColumnBiomeContainer extends ChunkBiomeContainer {
+    private final ChunkBiomeContainer[] containers;
+    private final int minCubeY;
+    private int fallBackIdx = -1;
+
+    public ProtoColumnBiomeContainer(IBigCube[] cubes, int xSectionOffset, int zSectionOffset) {
+        super(null, new ProtoColumnHeightAccess(cubes), (Biome[]) null);
+        if (!((CubicLevelHeightAccessor) cubes[0]).isCubic()) {
+            throw new UnsupportedOperationException("Calling a cube class in a non cubic world.");
+        }
+
+        this.minCubeY = cubes[0].getCubePos().getY();
+
+        ChunkBiomeContainer[] containers = new ChunkBiomeContainer[cubes.length];
+        for (int yCube = 0; yCube < cubes.length; yCube++) {
+            ChunkBiomeContainer biomeContainer = cubes[yCube].getBiomes();
+
+            if (biomeContainer instanceof CubeBiomeContainer) {
+                if (fallBackIdx == -1) {
+                    fallBackIdx = yCube;
+                }
+
+                containers[yCube] = ((CubeBiomeContainer) biomeContainer).getContainerForColumn(xSectionOffset, zSectionOffset);
+            } else {
+                if (biomeContainer != null) {
+                    throw new UnsupportedOperationException("Attempted to set a non cube column biome container, this operation is not supported. Attempted to set a biome container of "
+                        + "type: " + biomeContainer.getClass().getSimpleName());
+                }
+            }
+        }
+        this.containers = containers;
+    }
+
+    protected ProtoColumnBiomeContainer(IdMap<Biome> idMap, LevelHeightAccessor levelHeightAccessor, Biome[] biomes) {
+        super(idMap, levelHeightAccessor, biomes);
+        throw new UnsupportedOperationException("We are Using the wrong constructor!");
```

Comment:
Ideally these messages should tell the user which constructor to use, thus avoiding confusion/unnecessary work.